### PR TITLE
[ISSUE-228][FEATURE] Add a period local storage cleaner thread

### PR DESCRIPTION
**Pull request by sfwang218**
### What changes were proposed in this pull request?
Add a period local storage cleaner thread to check the leak shuffle data on disks

### Why are the changes needed?
Because we don't use strict lock strategy, we will delete some shuffle data normally. If we use strict lock strategy, the performance will be bad. So we want to add a period local storage cleaner thread. The thread examine which application is in the disk and judge whether the app exists the memory data structure ShuffleTaskManager#shuffleTaskInfos. If memory data structure don't exist, we will delete the data in the disk, with issue #282 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
tests were added

### CODE REVIEW

File: storage/src/main/java/org/apache/uniffle/storage/handler/api/ShuffleDeleteHandler.java [hunk 1]
```diff
@@ -25,4 +25,11 @@ public interface ShuffleDeleteHandler {
    * @param appId ApplicationId for delete
    */
   void delete(String[] storageBasePaths, String appId, String user);
+
+  /**
+   * Delete shuffle data with appId
+   *
+   * @param appId ApplicationId for delete
+   */
+  void delete(String[] storageBasePaths, String appId);
```
Comment (jerqi) [inline storage/src/main/java/org/apache/uniffle/storage/handler/api/ShuffleDeleteHandler.java:34]: Why do we need this interface? Could we reuse `delete(String[] storageBasePaths, String appId, String user)`?

File: server/src/main/java/org/apache/uniffle/server/ShuffleServerConf.java [hunk 2]
```diff
@@ -308,6 +308,12 @@ public class ShuffleServerConf extends RssBaseConf {
       .defaultValue(60 * 1000L)
       .withDescription("The timeout of the cache which record the mapping information");
 
+  public static final ConfigOption<Long> SERVER_LEAK_SHUFFLE_DATA_CHECK_INTERVAL = ConfigOptions
```
Comment (jerqi) [inline server/src/main/java/org/apache/uniffle/server/ShuffleServerConf.java:311]: Could you add this config option in our document https://github.com/apache/incubator-uniffle/blob/master/docs/server_guide.md ？

**Comment by jerqi** [comment]
企业微信截图_4c4d86f5-edf1-45fa-a8a1-e1c3e8284aa8
We should fix the code style. you can see https://github.com/apache/incubator-uniffle/actions/runs/3540626983/jobs/5943958492

**Comment by sfwang218** [comment]
OK I will fix it

**Review by jerqi — APPROVED** [review]
LGTM, wait for CI

**Comment by jerqi** [comment]
Merged. Thanks @sfwang218 , It's great work!

**Pull request merged**